draft-ietf-quic-manageability.md

```diff
@@ -809,20 +816,83 @@ off the connection ID may therefore cause undetectable and unrecoverable loss
 of state in the middle of a connection. Use of connection ID specifically
 discouraged for NAT applications.
 
-## Passive Network Performance Measurement and Troubleshooting
+If a NAT hits an operational limit, it is recommended to rather drop the
+initial packets of a flow (see also {{sec-filtering}}),
+which potentially triggers a fallback to TCP, than using the connection ID to
+muliplex mulitple connections on the same IP address port pair which risks
+breakage later on connection ID change.
+
+## Address Rewriting to Ensure Routing Stability
+
+If the routers or switches in the server infrastructure rely on address-port
```

Comment:
I don't think this section captures the purpose of the text it's replacing.

1.  Whether or not I have QUIC-LB, if there's some part of the infrastructure that wants stable 4tuples,  I could try to use the CID to catch NAT rebindings and just rewrite the source address to mask the change from the infrastructure. Yes, this will break if CIDs change; but even if the CID stays the same it will also break the security mechanisms that protect attacks related to NAT rebinding. This is the point of the original text.

2. Whether or not I have QUIC-LB, if the CID and 4tuple change there is no way to maintain the original source and IP port for the connection.

Reply:
Regarding your point 1: I think this is covered by 
"this approach is not recommended to be used with NAT functions, as hiding address
 changes from a server prevents the server from validating new addresses,
 which would facilitate amplification attacks"
What's missing?

Regarding you point 2, I would think it is also possible to use similar mechanisms as described in the LB doc to not only encode with server but also the actual connection, no?

Reply:
This is the text from the other rewrite attempt:

```
Concealing client address changes in order to simplify operational routing
 issues will mask important signals that drive security mechanisms, and
 therefore opens QUIC up to various attacks.

While QUIC's migration capability makes it possible for an server to survive
 address changes, this does not work if the routers or switches in the server
 infrastructure rely on address-port 4-tuple as a NAT rebinding or address
 migration will cause packets to be delivered to the wrong server. {{QUIC_LB}}
 addresses this problem proposing a QUIC extension to allow server-load balancer
 coordination to routable CIDs.

An alternative, potentially simpler approach would seem to be the use of NAT
 in front of such an infrastructure setup. However, hiding information about the
 change of the IP address or port conceals important, and security relevant
 information from QUIC endpoints, and as such would facilitate amplification
 attacks (see section 9 of {{QUIC-TRANSPORT}}).

 An NAT that bleaches peer address changes, hinders the other end
 to detect and mitigate these attacks and verify connectivity to the
 new address based on QUIC PATH_CHALLENGE and PATH_RESPONSE frames.
```

Or is this a better starting point?

Reply:
I like the other rewrite text better, yes.

On point 1: what I'm saying is that QUIC-LB doesn't enter into it.

On point 2: yes, that is theoretically possible. However, I don't want to suggest in any way that we make CIDs absolutely linkable. Exposing the server mapping is bad enough!